libbeat/outputs/kafka/config.go

```diff
@@ -58,5 +82,13 @@ func (c *kafkaConfig) Validate() error {
 		return fmt.Errorf("compression mode '%v' unknown", c.Compression)
 	}
 
+	if _, ok := kafkaVersions[c.Version]; !ok {
+		return fmt.Errorf("unknown/unsupported kafka version '%v'", c.Version)
+	}
+
+	if c.Username != "" && c.Password == "" {
+		return fmt.Errorf("password must not be empty when SASL user is configured")
```

Comment:
I would rephrase this to "empty password. Password config option must be set when Username is configured."